### ISSUE TITLE
Electricity Is Cheapest Looking X Hours Forward

### ISSUE THREAD

**Issue by 02curls** [rnovacek/homeassistant_cz_energy_spot_prices#36]
dobrý den,

těmito dvěma automatizacemi zhruba optimalizuji vybíjení FV baterky v hodiny dne, které jsou spotově dražší:
**When Current Spot Electricity Hour Order is >12  ALLOW DIScharge
When Current Spot Electricity Hour Order is <13  DISALLOW DIScharge**

Jenže narážím na půlnoční "nesouvislé" řazení hodin druhého dne, které teď vede k tomu, že např. ve v příloze naznačené situaci automatizace od 22 do 24 hodin prvního dne vybíjení zakazuje, i přestože spotové ceny následujících ranních hodin nového dne stále klesají a tak stálo za to, baterku vybíjet.

nešlo by vytvořit nové entity jako např. **Electricity Is Cheapest Looking X Hours Forward**,  **Electricity Is Most Expensive Looking X Hours Forward**, které by braly v potaz ceny dalšího dne ?

pardon pokud lze věc řešit stávajícími entitami, prosím eventuálně o nasměrování

díky moc

<img width="685" alt="Screenshot 2023-10-17 at 8 50 15" src="https://github.com/rnovacek/homeassistant_cz_energy_spot_prices/assets/107803294/43a48f90-c9d7-4634-9016-0e1341f4eb8e">

**Comment by hipik634** [comment]
ahoj, prosímtě můžeš mi říct, jak pracuješ s těmi Current Spot Electricity Hour Order, já se trápím s tím, že hodnoty jsou v atributech a nevím, jak na to. Prosím příklad od Tebe z HA, díky

**Comment by johny-mnemonic** [comment]
@mrpohoda Řeším teď stejný problém:
> "nalézt nejlevnější hodinu nebo více po sobě jdoucích hodin během noci (např. od 20 večer do 8 rána)"

Podařilo se najít nějaké řešení s pomocí této integrace?

Mě přijde, že nejlepší by bylo vygenerovat něco jako `current_spot_electricity_hour_order` pro daný interval (noc), protože pak by se to dalo používat v automatizacích univerzálně a hlavně dynamicky. Počet nejlevnějších hodin, které potřebuje jedna automatizace bývá jiný, než kolik jich potřebuje jiná a mění se to i každý den u stejné automatizace. Další variantou je udělat si template senzory na 2,3,4 nejlevnější hodiny v noci, jak ukázal @shaman79 v https://github.com/rnovacek/homeassistant_cz_energy_spot_prices/issues/40, ale ten `hour_order` senzor mi přijde jako elegantnější řešení, obzvlášť pokud nevadí nespojitá dotace hodin.

**Comment by mrpohoda** [comment]
> @mrpohoda Řeším teď stejný problém:
> 
> > "nalézt nejlevnější hodinu nebo více po sobě jdoucích hodin během noci (např. od 20 večer do 8 rána)"
> 
> Podařilo se najít nějaké řešení s pomocí této integrace?

Řeším to pomocí funkce v Node Red. V 15 hodin odpoledne si spočítám začátek a konec nabíjení pro noc (offpeak), v 7 hodin ráno spočítám nejlevnější nabíjecí čas pro den (peak). Kód funkce [tady](https://pastebin.com/je79RXh4)

![Screenshot 2024-03-21 at 13 09 40](https://github.com/rnovacek/homeassistant_cz_energy_spot_prices/assets/922916/a7f8f613-b974-459c-883d-5f806286d22a)

**Comment by johny-mnemonic** [comment]
@mrpohoda Díky, ale Node Red nepoužívám. Ještě zkusím bojovat v HA :-)

**Comment by johny-mnemonic** [comment]
@Policanek už se to tu řešilo na několika místech. Nejvíce je asi [zde](https://github.com/rnovacek/homeassistant_cz_energy_spot_prices/issues/40).
Ve zkratce, dokud něco takového nezačne podporovat sama integrace, tak nejlepší řešení je šablona.

**Comment by rnovacek** [comment]
Ano, v současné době je nejlepší řešení [šablona](https://github.com/rnovacek/homeassistant_cz_energy_spot_prices?tab=readme-ov-file#find-cheapest-hours-in-selected-interval).
